lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -40,6 +40,14 @@ def deploy_failed?
       @pods.present? && @pods.all?(&:deploy_failed?)
     end
 
+    def failure_message
+      @pods.map(&:failure_message).compact.uniq.join("\n")
```

Comment:
No, this is the custom part of the message that optionally gets printed before the logs/events/etc. If 500 pods fail to start and the `Container#doom_reasons` aren't all the same, it would print however many variations there were. I'd expect that there would be very little variation though, given that the dynamic info in doom reasons is just exit status code, image name and container config problem (e.g. missing secret name).